lib/drivers/abbottFreeStyle.js

```diff
@@ -0,0 +1,562 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2014, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+var _ = require('lodash');
+var async = require('async');
+var sundial = require('sundial');
+var struct = require('../struct.js')();
+
+module.exports = function (config) {
+  var cfg = _.clone(config);
+  cfg.deviceData = null;
+  var serialDevice = config.deviceComms;
+
+  var STX = 0x02;
+  var ETX = 0x03;
+  var CR = 0x0D;
+  var LF = 0x0A;
+
+  var dateTimePattern = '([A-Za-z ]{4}) ([0-3][0-9]) ([0-9]{4}) ' +
+                        '([012][0-9]):([0-5][0-9])(?::([0-6][0-9]))*';
+  function parseDateTime(match, offset) {
+    var fmt = 'MMMM DD YYYY HH:mm:ss';
+    var utc = sundial.parseAndApplyTimezone(match[0], fmt, cfg.timezone).toISOString();
+    var dev = sundial.formatDeviceTime(utc);
```

Comment:
`deviceTime` should be our preserved record of the device's local time; that's not what you're generating here since the timezone has already been applied.

The way I'd do it with the functions we currently provide in sundial is the following (note that `parseAndApplyTimezone` does not require a timezone; if none is provided, it simply parses from the format...perhaps the method name is not the most intuitive here):
```
var parsed = sundial.parseAndApplyTimezone(match[0], fmt);
var dev = sundial.formatDeviceTime(parsed);
var utc = sundial.applyTimezone(parsed, cfg.timezone).toISOString();
```